demos/compute_demo.py

```
@@ -33,8 +34,44 @@ from libcloud.providers import get_driver
 
 from pprint import pprint
 
+def get_demo_driver(provider_name='RACKSPACE', *args, **kwargs):
+    """An easy way to play with a driver interactively.
+
+    # Load credentials from secrets.py:
+    >>> from compute_demo import get_demo_driver
+    >>> driver = get_demo_driver('RACKSPACE')
+
+    # Or, provide credentials:
+    >>> from compute_demo import get_demo_driver
+    >>> driver = get_demo_driver('RACKSPACE', 'username', 'api_key')
+    # Note that these parameters vary by driver ^^^
+
+    # Do things like the demo:
+    >>> driver.load_nodes()
+    >>> images = driver.load_images()
+    >>> sizes = driver.load_sizes()
+    
+    # And maybe do more than that:
+    >>> node = driver.create_node(
+            name='my_first_node',
+            image=images[0],
+            size=sizes[0],
+        )
+    >>> node.destroy()
+    """
+    provider_name = provider_name.upper()
+
+    DriverClass = get_driver(getattr(Provider, provider_name))
+
+    if not args:
```

I would throw an exception in case both `args` and `kwargs` are None.

The exception message should say something like: "Missing credentials / arguments for provider the  X. Please edit secrets.py and put the credentials there."